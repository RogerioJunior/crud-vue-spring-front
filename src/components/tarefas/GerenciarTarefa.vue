<script>
    import gerenciarTarefaBusiness from "../../business/tarefaBusiness/gerenciarTarefaBusiness";
    import router from '../../router'

    export default {
        title () {
            return this.tarefaDto ? "Editar Tarefa" : "Adicionar Tarefa";
        },
        components: {

        },

        props: [
            'tarefaDto'
        ],

        data () {
            return {
                form: {
                    nome: "",
                    descricao: "",
                    nota: ""
                }
            }

        },

        beforeMount(){
            if (this.tarefaDto) {
                this.$set(this.form, 'nome', this.tarefaDto.nome);
                this.$set(this.form, 'descricao', this.tarefaDto.descricao);
                this.$set(this.form, 'nota', this.tarefaDto.nota);

            }
        },

        computed:{

            setTitulo () {
                return this.tarefaDto ? "Editar Tarefa" : "Adicionar Tarefa";
            },
        },

        methods: {

            async salvar () {
                if (this.tarefaDto) {
                    await gerenciarTarefaBusiness.editarTarefa(this.tarefaDto.id, this.form);
                }else{
                    await gerenciarTarefaBusiness.salvarTarefa(this.form);
                }
                router.push({name: 'Principal'})
            }
        }
    }
</script>

<template>

    <div>
        <h2>{{setTitulo}}</h2>
        <form @submit.prevent="">
            <div class="form-group">
                <label for="nome">Nome</label>
                <input class="form-control" id="nome" v-model="form.nome" required/>
            </div>
            <div class="form-group">
                <label for="descricao">Descrição</label>
                <textarea class="form-control" id="descricao" rows="5" v-model="form.descricao"></textarea>
            </div>
            <div class="form-group">
                <label for="nota">Nota</label>
                <input type="number" class="form-control" id="nota" v-model="form.nota"/>
            </div>
            <div class="form-group gerenciar">
                <button type="submit" class="btn btn-primary" @click="salvar()">Salvar</button>

                <router-link :to="{name: 'Principal'}">
                    <!-- <router-link to="/">-->
                    <button type="button" class="btn btn-default"> Cancelar</button>
                </router-link>
            </div>
        </form>
    </div>
</template>

<style lang="scss" src="../../css/tarefas.scss"></style> 
 