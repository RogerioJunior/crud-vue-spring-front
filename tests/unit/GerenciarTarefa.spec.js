// import { mount } from '@vue/test-utils' *mount monta o componente e carrega os filhos dele tambem, podendo gerar um efeito colateral
import { shallowMount } from '@vue/test-utils'
import GerenciarTarefa from '@/components/tarefas/GerenciarTarefa.vue'
import Vue from 'vue'
import VueRouter from 'vue-router'

Vue.use(VueRouter);
// quando o jest chega no gerenciarTarefa ( monta o componente de forma isolada),
// ele não chega no main.js, assim não reconhece/nao sabe o que fazer com o <router-Link>,
// temos que importar o que vem de fora ex. router, vuelidate

describe('Gerenciar Tarefa', () => {

  let wrapper;

  beforeAll(() => { // executa o metodo e depois o callback antes dos testes
    wrapper = shallowMount(GerenciarTarefa);
  });

  test('As propriedades corretas estão presentes no estado do componente', () => {
    const expected = ['form'];
    const received = Object.keys(wrapper.vm.$data);
    expect(received).toEqual(expected)
  });


  test(' Componente é uma instancia do Vue', () => {
    const wrapper = shallowMount(GerenciarTarefa);
    expect(wrapper.isVueInstance()).toBeTruthy()
  });


  test('renders props.msg when passed', () => {
      const msg = "Adicionar Tarefa Nome  Descrição  Nota  Salvar  Cancelar";
      const wrapper = shallowMount(GerenciarTarefa, {
        propsData: { msg }
      });
      expect(wrapper.text()).toMatch(msg)
    });



});




// npm t -- --watch ( roda apenas o teste que foi alterado, criado em um commit)
