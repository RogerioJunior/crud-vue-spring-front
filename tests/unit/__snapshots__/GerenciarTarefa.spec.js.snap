// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gerenciar Tarefa componente está renderizando 1`] = `
<div>
  <h2>Adicionar Tarefa</h2>
  <form>
    <div class="form-group"><label for="nome">Nome</label> <input id="nome" required="required" class="form-control"></div>
    <div class="form-group"><label for="descricao">Descrição</label> <textarea id="descricao" rows="5" class="form-control"></textarea></div>
    <div class="form-group"><label for="nota">Nota</label> <input type="number" id="nota" class="form-control"></div>
    <div class="form-group gerenciar"><button type="submit" class="btn btn-primary">Salvar</button>
      <router-link-stub to="[object Object]" tag="a" event="click"><button type="button" class="btn btn-default"> Cancelar</button></router-link-stub>
    </div>
  </form>
</div>
`;
